components/recovery/src/service.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_rocks::RocksEngine;
+use engine_traits::{Engines, RaftEngine};
+
+use futures::sink::SinkExt;
+
+use grpcio::{
+    RpcContext,
+    UnarySink,
+    RequestStream,
+    ClientStreamingSink,
+    ServerStreamingSink,
+
+};
+use kvproto::{
+    recoverdatapb::{*},
+};
+use raftstore::{
+    store::fsm::RaftRouter,
+};
+
+/// Service handles the recovery messages from backup restore.
+#[derive(Clone)]
+pub struct RecoveryService<ER: RaftEngine> {
+    engines: Engines<RocksEngine, ER>,
+    router: RaftRouter<RocksEngine, ER>,
+}
+
+#[warn(dead_code)]
+impl<ER: RaftEngine> RecoveryService<ER> {
+    /// Constructs a new `Service` with `Engines`, a `RaftStoreRouter` and a `thread pool`.
+    pub fn new(
+        engines: Engines<RocksEngine, ER>,
+        router: RaftRouter<RocksEngine, ER>,
```

Comment:
How to get a router? The router is only available after tikv server is started.

I suggest to start the work from `main` function to make it more clear.